core/impl/src/main/java/com/blazebit/persistence/impl/function/datetime/yearweek/H2YearWeekFunction.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2014 - 2019 Blazebit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.blazebit.persistence.impl.function.datetime.yearweek;
+
+/**
+ * @author Jan-Willem Gmelig Meyling
+ * @since 1.4.0
+ */
+public class H2YearWeekFunction extends YearWeekFunction {
+
+    public H2YearWeekFunction() {
+        super("(select CONCAT(extract(year from DATE_TRUNC('week', x.y)), '-', iso_week(x.y)) from (values (?1)) x(y))");
```

Comment:
Unfortunately, H2 doesn't support that syntax. In DB2 this works by using `from lateral (values (?1))` but I doubt this will work here. Can you think of any other SQL that might be able to produce the same result while rendering the template argument only once? If not, we will have to bail if encountering a parameter.